### PR TITLE
Checkpoint class rob

### PR DESCRIPTION
**Pull request by roblim**
- add logic to Checkpoint class to handle checkpoint config processing at runtime (templates, runtime kwargs)

### CODE REVIEW

File: great_expectations/checkpoint/checkpoint.py [hunk 1]
```diff
@@ -88,28 +104,111 @@ def _get_batches_to_validate(self, batches):
 
 class Checkpoint(object):
     def __init__(
-        self, data_context, name: str, validation_operator_name: str, validators: list,
+        self,
```
Comment (alexsherstinsky) [inline great_expectations/checkpoint/checkpoint.py:107]: Inheriting from `object` is no longer favored in the Python best practices.

File: great_expectations/checkpoint/checkpoint.py [hunk 2]
```diff
@@ -88,28 +104,111 @@ def _get_batches_to_validate(self, batches):
...
+
+                self._substituted_config = substituted_config
+                return substituted_config
+            else:
+                template_config = self.data_context.get_checkpoint(
+                    checkpoint_name=template_name, return_config=True
+                )
 
-        return validators
+                if float(template_config.config_version) != float(
```
Comment (alexsherstinsky) [inline great_expectations/checkpoint/checkpoint.py]: Will this work without the `cast`?..

File: great_expectations/checkpoint/checkpoint.py [hunk 3]
```diff
@@ -88,28 +104,111 @@ def _get_batches_to_validate(self, batches):
...
+            "run_name_template": run_name_template,
+            "expectation_suite_name": expectation_suite_name,
+            "batch_request": batch_request,
+            "action_list": action_list,
+            "evaluation_parameters": evaluation_parameters,
+            "runtime_configuration": runtime_configuration,
+            "validations": validations,
+            "profilers": profilers,
+        }
+        substituted_config = self.get_substituted_config(runtime_kwargs=runtime_kwargs)
```
Comment (alexsherstinsky) [inline great_expectations/checkpoint/checkpoint.py:215]: This looks like an abrupt ending to the `run` method -- isn't there anything more after line 214?  Or is this just GitHub now showing me more?..

Comment (roblim) [reply great_expectations/checkpoint/checkpoint.py:215]: Sorry, I should have added a TODO comment. Will finish up run method next

**Review by alexsherstinsky — APPROVED** [review]
LGTM as long as the comments are either followed up upon or rationalized with an explanation.

File: great_expectations/checkpoint/checkpoint.py (hunk 2 above)
Comment (roblim) [reply great_expectations/checkpoint/checkpoint.py]: Oh, you're totally right.

**Pull request merged**